### PR TITLE
bootstrap: Add llvm-has-rust-patches target option

### PR DESCRIPTION
**Pull request by tmandry**
This is so you can check out an upstream commit in src/llvm-project and
have everything just work.

This simplifies the logic in `is_rust_llvm` a bit; it doesn't need to
check for download-ci-llvm because we would have already errored if both
that and llvm-config were specified on the host platform.

### CODE REVIEW

File: config.toml.example [hunk 1]
```diff
@@ -666,6 +666,10 @@ changelog-seen = 2
 # target.
 #llvm-config = <none> (path)
 
+# Override detection of whether this is a Rust-patched LLVM. This would be used
+# in conjunction with either an llvm-config or build.submodules = false.
+#llvm-is-vanilla = if llvm-config { true } else { false }
```
Comment (Mark-Simulacrum) [inline config.toml.example:671]: I feel like llvm-has-rust-patches here is probably clearer naming.

File: src/bootstrap/lib.rs [hunk 2]
```diff
@@ -854,12 +855,9 @@ impl Build {
...
     /// If no custom `llvm-config` was specified then Rust's llvm will be used.
     fn is_rust_llvm(&self, target: TargetSelection) -> bool {
-        if self.config.llvm_from_ci && target == self.config.build {
-            return true;
-        }
-
         match self.config.target_config.get(&target) {
-            Some(ref c) => c.llvm_config.is_none(),
+            Some(Target { llvm_is_vanilla: Some(is_vanilla), .. }) => !is_vanilla,
+            Some(Target { llvm_config, .. }) => llvm_config.is_none(),
```
Comment (Mark-Simulacrum) [inline src/bootstrap/lib.rs:860]: Seems like this is a behavior change -- previously llvm_from_ci would return true here, but I think this returns false now?

The way CI LLVM works *is* by setting up llvm-config.

Comment (tmandry) [reply src/bootstrap/lib.rs:860]: Oh I missed that it set llvm-config. Fixed.

**Pull request merged**